shell-ui/src/auth/permissionUtils.js

```diff
@@ -0,0 +1,40 @@
+//@flow
+import type { Options, TranslationAndGroups } from '../index';
+
+export const isEntryAccessibleByTheUser = (
+  [path, translationAndGroup]: [string, TranslationAndGroups],
+  userGroups: string[],
+): boolean => {
+  return (
+    (translationAndGroup.groups &&
+      translationAndGroup.groups
+        .map((group) => userGroups.includes(group))
+        .reduce(
+          (aggregatedPredicates, current) => aggregatedPredicates && current,
+          true,
+        )) ||
+    !translationAndGroup.groups
+  );
+};
+
+export const getAccessiblePathsFromOptions = (
+  options: Options,
+  userGroups: string[],
+): string[] => {
+  return [...Object.entries(options.main), ...Object.entries(options.subLogin)]
+    //$FlowIssue - flow typing for Object.entries incorrectly typing values as [string, mixed] instead of [string, TranslationAndGroups]
+    .filter((entry: [string, TranslationAndGroups]) =>
+      isEntryAccessibleByTheUser(entry, userGroups),
+    )
+    .map(([path]) => path);
```

Comment:
This is already a builtin:

```suggestion
    .flat();
```

Reply:
I can't use flat here because I actually want to destructure the first entry of Object entries to retrieve only the `path`.

Reply:
Ah right, read too fast :+1: